### ISSUE TITLE
check_ceph_mon always show "MON OK"

### ISSUE THREAD

**Issue by kalloa** [valerytschopp/ceph-nagios-plugins#10]
Hi!
First of all, thanks for this plugin.
I have ceph cluster with disabled cephx and a ceph monitor down name node2:
```
root@node1:~# ceph mon stat
e3: 3 mons at {node1=172.16.1.1:6789/0,node2=172.16.1.2:6789/0,node3=172.16.1.3:6789/0}, election epoch 522, quorum 0,2 node1,node3
```
and also no ping replo to node2:
```
root@node1:~# ping node2
PING node2 (172.16.1.2) 56(84) bytes of data.
^C
--- node2 ping statistics ---
2 packets transmitted, 0 received, 100% packet loss, time 1007ms
```
but the plugin says ok
```
root@node1:~# ./ceph-nagios-plugins-master/src/check_ceph_mon -I node2 -H node2
MON OK
```
What can be wrong?

**Comment by valerytschopp** [comment]
Hi kolloa,

What is the ouput of 'ceph mon dump' on node1 or node2 ?

**Comment by valerytschopp** [comment]
And can you please also try on your node1 and node2:

`ceph mon_status`

**Comment by kalloa** [comment]
Hi valerytschopp!
I can't try this commands on node2, becaus its momentarily down(maybe next week will be up),
ceph mon dump on node1:
```
root@node1:~#  ceph mon dump
dumped monmap epoch 3
epoch 3
fsid 89b36093-e011-40c2-9993-94529d8b09d9
last_changed 2015-07-03 16:12:37.891730
created 0.000000
0: 172.16.1.1:6789/0 mon.node1
1: 172.16.1.2:6789/0 mon.node2
2: 172.16.1.3:6789/0 mon.node3
```
```
root@node1:~# ceph mon_status
2015-07-21 16:57:52.123241 7f9d5447d700  0 -- :/1030424 >> 172.16.1.2:6789/0 pipe(0x7f9d500261a0 sd=3 :0 s=1 pgs=0 cs=0 l=1 c=0x7f9d5002a440).fault
{"name":"node3","rank":2,"state":"peon","election_epoch":524,"quorum":[0,2],"outside_quorum":[],"extra_probe_peers":[],"sync_provider":[],"monmap":{"epoch":3,"fsid":"89b36093-e011-40c2-9993-94529d8b09d9","modified":"2015-07-03 16:12:37.891730","created":"0.000000","mons":[{"rank":0,"name":"node1","addr":"172.16.1.1:6789\/0"},{"rank":1,"name":"node2","addr":"172.16.1.2:6789\/0"},{"rank":2,"name":"node3","addr":"172.16.1.3:6789\/0"}]}}
```

**Comment by valerytschopp** [comment]
Thanks!

I see the problem: 
*  `check_ceph_mon` parses the output of `ceph mon dump`
* but the output doesn't report the up/down state of the mon

Basically, the check doesn't check if a monitor is down :(

**Comment by valerytschopp** [comment]
So the `check_ceph_mon` should parse `ceph mon_status` or `ceph quorum_status`

Can you please send me the output of `ceph quorum_status` on nod1?

**Comment by valerytschopp** [comment]
This should help to check for down monitors http://ceph.com/docs/master/rados/troubleshooting/troubleshooting-mon/#understanding-mon-status

**Comment by kalloa** [comment]
Node2 monitor stil down:
```
root@node1:~# ceph quorum_status
{"election_epoch":636,"quorum":[0,2],"quorum_names":["node1","node3"],"quorum_leader_name":"node1","monmap":{"epoch":3,"fsid":"89b36093-e011-40c2-9993-94529d8b09d9","modified":"2015-07-03 16:12:37.891730","created":"0.000000","mons":[{"rank":0,"name":"node1","addr":"172.16.1.1:6789\/0"},{"rank":1,"name":"node2","addr":"172.16.1.2:6789\/0"},{"rank":2,"name":"node3","addr":"172.16.1.3:6789\/0"}]}}
```
```
root@node1:~# ceph mon_status
{"name":"node3","rank":2,"state":"peon","election_epoch":636,"quorum":[0,2],"outside_quorum":[],"extra_probe_peers":[],"sync_provider":[],"monmap":{"epoch":3,"fsid":"89b36093-e011-40c2-9993-94529d8b09d9","modified":"2015-07-03 16:12:37.891730","created":"0.000000","mons":[{"rank":0,"name":"node1","addr":"172.16.1.1:6789\/0"},{"rank":1,"name":"node2","addr":"172.16.1.2:6789\/0"},{"rank":2,"name":"node3","addr":"172.16.1.3:6789\/0"}]}}
```
```
root@node1:~# ceph -s
2015-07-28 11:08:58.336307 7fb33c7d6700  0 -- :/1008492 >> 172.16.1.2:6789/0 pipe(0x7fb338025e00 sd=3 :0 s=1 pgs=0 cs=0 l=1 c=0x7fb33802a0a0).fault
    cluster 89b36093-e011-40c2-9993-94529d8b09d9
     health HEALTH_WARN
            1 mons down, quorum 0,2 node1,node3
     monmap e3: 3 mons at {node1=172.16.1.1:6789/0,node2=172.16.1.2:6789/0,node3=172.16.1.3:6789/0}
            election epoch 636, quorum 0,2 node1,node3
     osdmap e137961: 11 osds: 11 up, 11 in
      pgmap v754951: 768 pgs, 3 pools, 184 GB data, 642 kobjects
            318 GB used, 33659 GB / 33977 GB avail
                 768 active+clean
  client io 13864 B/s wr, 0 op/s
```

**Comment by kalloa** [comment]
If I stop only the monitor service on node2 the plugin check_ceph_health works like a charm and show the monitor is down error: HEALTH WARNING: 1 mons down, quorum 0,2 node1,node3

**Comment by valerytschopp** [comment]
Thanks a lot, the output of `ceph quorum_status` clearly shows that the `node2` is not in the quorum, and should be down or inactive...

I'll update the plugin to parse this output and give a correct warning

**Comment by valerytschopp** [comment]
The new refactored `check_ceph_mon` is available a98bfcd90d45215e3996fa4286fc3cd59f840d73

**Comment by kalloa** [comment]
Thanks valerytschopp!
Now the new version works fine.
